pkg/build/builder/common.go

```diff
@@ -201,15 +219,31 @@ func retryBuildStatusUpdate(build *buildapi.Build, client client.BuildInterface,
 		latestBuild.Status.Output.To = build.Status.Output.To
 		latestBuild.Status.Stages = build.Status.Stages
 
-		if _, err := client.UpdateDetails(latestBuild); err != nil {
-			return err
+		_, err = client.UpdateDetails(latestBuild)
+		netErr, ok := err.(net.Error)
+
+		switch {
+		case err == nil:
+			shouldNotRetry = true
+			returnErr = nil
+		case errors.IsConflict(err):
+			latestBuild = nil
+			shouldNotRetry = false
+			returnErr = nil
+		case ok && netErr.Temporary():
+			shouldNotRetry = false
+			returnErr = nil
```

Comment:
not sure why you're setting returnError to nil in these two cases.  If we timeout because we kept getting the same temporary error over and over again (or a conflict error over and over again), i think we'd still want to know that.